### ISSUE TITLE
Add support for mqtt protocol

### ISSUE THREAD

**Issue by actuaIIy** [httptoolkit/httptoolkit#254]
Please add support for mqtt websockets, pretty sure quite some people would want this implemented.

**Comment by actuaIIy** [comment]
![image](https://user-images.githubusercontent.com/84027814/159044361-b097a98d-162c-42b8-be92-55a1df4a3737.png)
this is what I currently see upon a mqtt connection

**Comment by andreavalle** [comment]
I'm also interested in MQTT support, and in general anyone testing IoT apps would require MQTT support.
MQTT is a publish and subscribe protocol.
Typically an IoT device would publish some real time data (e.g. a temperature from a sensor) and an application would subscribe to a topic to receive the temperature values as they are published by the sensor.
Having said that, would you consider to make a plan to support MQTT in HTTP Toolkit?

**Comment by pimterry** [comment]
> Having said that, would you consider to make a plan to support MQTT in HTTP Toolkit?

Absolutely, I'd be happy to have it supported! That said, it's quite a bit of work, and the todo list is already very long.

If you're interested in this, pull requests are very welcome and do :+1: on this issue to vote for it.

If you'd like to help with some PRs to move this forward, the first step is to add support to detect MQTT in [httpolyglot](https://github.com/httptoolkit/httpolyglot/) which looks at the first bytes of a connection to work out which protocol it is (currently handling TLS, HTTP/1 & no-negotiation HTTP/2 all on the same port). I'm not sure what the initial packet of an MQTT connection looks like, but researching and detecting that would all us to independently handle those.

Once that's working, the next step is building a plugin for [Mockttp](https://github.com/httptoolkit/mockttp) (which uses Httpolyglot to handle HTTP, and provides a base supporting plugins for other protocols) so we can integrate MQTT into the existing HTTP-based interception framework. [MockRTC](https://github.com/httptoolkit/mockrtc/) is an example of this for WebRTC, although the plugin framework would probably need a little tweaking for this case.

That's the main hard bit. When that's done, the rest of the work becomes clearer and easier: the new library (let's call it 'MockQTT') can announce events for new connections and messages in each direction (exactly as Mockttp does for HTTP), the [UI model](https://github.com/httptoolkit/httptoolkit-ui/blob/master/src/model/events/events-store.ts) would subscribe to those (as it does for HTTP) to collect all the MQTT data, and then we'd need [UI components](https://github.com/httptoolkit/httptoolkit-ui/tree/master/src/components/view) (mainly just reusing the WebSockets UI I expect) to show the connection and the stream of messages within it.

If you're interested in this, feel free to look into that and open issues on the relevant repos if you have any questions. If you're interested but you don't have time for a PR, please to :+1: on this issue anyway, since that's one of the main ways we track priorities for what to implement next.

**Comment by jewelsonn** [comment]
Hey This may help you in some way to implement  MQTT https://github.com/mautrix/facebook/blob/master/maufbapi/mqtt/otclient.py

**Comment by nutterthanos** [comment]
note for reference python has this paho.mqtt https://pypi.org/project/paho-mqtt/
or this:
https://github.com/mqtt-tools/mqttdecode
